test/gtest/inline_vector_basic_ops.cpp

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+
+#include <numeric>
+
+#include <miopen/inline_vector.hpp>
+#include <miopen/tensor.hpp>
+
+namespace {
+
+TEST(CPU_InlineVectorSizeAndAccumulate_NONE, Test)
+{
+    miopen::InlineVector<std::size_t, 5> in_v1{4, 2, 1};
+    std::vector<size_t> v1{4, 2, 1};
+
+    EXPECT_EQ(in_v1.size(), v1.size());
+
+    for(uint8_t i = 0; i < in_v1.size(); i++)
+    {
+        EXPECT_EQ(in_v1[i], v1[i]);
+    }
+
+    int sum_in_v1 = std::accumulate(in_v1.begin(), in_v1.end(), 0);
+    int sum_v1    = std::accumulate(v1.begin(), v1.end(), 0);
```

Comment:
I know that both vectors are `4, 2, 1`, but accumulating `size_t` to `int` usually is not a good idea (it's not just `int sum`, the most important is the last parameter of `std::accumulate`).
Probably it's simpler to make both vectors `int`.